File: backend/app/api/v1/schemas/account.py
"""Account 响应 schema。"""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class AccountBalance(BaseModel):
    total_equity_usd: str
    available_usd: str
    locked_usd: str
    positions_notional_usd: str
    realized_pnl_usd: str
    unrealized_pnl_usd: str
    currency: str
    updated_at: datetime

File: frontend/lib/api/account.ts
import { apiClient } from './client'

export type AccountBalance = {
  total_equity_usd: string
  available_usd: string
  locked_usd: string
  positions_notional_usd: string
  realized_pnl_usd: string
  unrealized_pnl_usd: string
  currency: string
  updated_at: string
}

export async function getAccountBalance(): Promise<AccountBalance> {
  const { data } = await apiClient.get<AccountBalance>('/account/balance')
  return data
}
